pulsar-client-cpp/lib/PartitionedConsumerImpl.cc

```diff
@@ -365,6 +365,7 @@ void PartitionedConsumerImpl::messageReceived(Consumer consumer, const Message&
         }
         messages_.push(msg);
         if (messageListener_) {
+            unAckedMessageTrackerPtr_->add(msg.getMessageId());
```

Comment:
Thanks for the work. Seems this is only related with (messageListener_ != null), And not covered by the test case added.